### DISCUSSION TITLE
Not recieving MO messages for with specific keywords set by SMSC

### DISCUSSION THREAD

**Discussion by elhananjair** [python-smpplib/python-smpplib#236] [Q&A]
Hello friends I am implementing SMPP client by binding to SMSC in bidirectional mode. The SMSC have provided my short codes for MT (Mobile Terminated) and MO (Mobile Originated). 
When I send sms to users I use MT short code as source address and it working fine. I can receive deliveries receipt for sent sms, and also get MO messages from user. But there are two keywords that the SMSC assigned for subscription to our campaign, like 'ok' for subscribing and 'stop' to leave the subscription. In my case I have tested with other keywords like 'hello' and my application do receive the message but when someone sends 'Ok' or 'Stop', I am not getting those particular messages.

Can someone say something about this? Thanks and also I don't know when I have to use the MO short codes in my application since there is not place to put it based on SMPP specification.
Thanks

**Comment by nelsonlekule2050** [comment]
i am trying to receive SMS (SMS MO) using set_message_received_handler but it does not seem to work.
getting the error below every time the request from SMSC hits my smpp client.
.
.
Traceback (most recent call last):
  File "test_receiver5.py", line 38, in <module>
    client.listen()
  File "/usr/local/lib/python3.6/site-packages/smpplib/client.py", line 401, in listen
    self.read_once(ignore_error_codes, auto_send_enquire_link)
  File "/usr/local/lib/python3.6/site-packages/smpplib/client.py", line 355, in read_once
    pdu = self.read_pdu()
  File "/usr/local/lib/python3.6/site-packages/smpplib/client.py", line 268, in read_pdu
    allow_unknown_opt_params=self.allow_unknown_opt_params,
  File "/usr/local/lib/python3.6/site-packages/smpplib/smpp.py", line 41, in parse_pdu
    new_pdu.parse(data)
  File "/usr/local/lib/python3.6/site-packages/smpplib/pdu.py", line 127, in parse
    self.parse_params(data[16:])
  File "/usr/local/lib/python3.6/site-packages/smpplib/command.py", line 315, in parse_params
    self.parse_optional_params(data[pos:])
  File "/usr/local/lib/python3.6/site-packages/smpplib/command.py", line 333, in parse_optional_params
    field = get_optional_name(type_code)
  File "/usr/local/lib/python3.6/site-packages/smpplib/command.py", line 71, in get_optional_name
    raise exceptions.UnknownCommandError('Unknown SMPP command code "0x%x"' % code)
smpplib.exceptions.UnknownCommandError: Unknown SMPP command code "0x606"
WARNING:smpp.Client.140556016443008:<smpplib.client.Client object at 0x7fd5bf6e5e80> was not closed
.
.
.
.
Below is my code.
.
import logging
import sys

import smpplib.gsm
import smpplib.client
import smpplib.consts



logging.basicConfig(level='DEBUG')
client = smpplib.client.Client('xx.xx.xx.xx', xx)
def getPdu(pdu):
    print(pdu.short_message.decode('utf-8'))
client.set_message_received_handler(getPdu)
client.connect()
client.bind_transceiver(system_id='xxxxx', password='xxxxx',system_type='smpp')
parts, encoding_flag, msg_type_flag = smpplib.gsm.make_parts('hello world \n')
for part in parts:
    pdu = client.send_message(
        source_addr_ton=smpplib.consts.SMPP_TON_NWSPEC,
        source_addr_npi=smpplib.consts.SMPP_NPI_ISDN,
        source_addr='1591',
        dest_addr_ton=smpplib.consts.SMPP_TON_NATNL,
        dest_addr_npi=smpplib.consts.SMPP_NPI_ISDN,
        # Make sure thease two params are byte strings, not unicode:
        destination_addr='xxxxxxxxx',
        short_message=part,
        data_coding=encoding_flag,
        esm_class=msg_type_flag,
        registered_delivery=True,
    )
    print(pdu.sequence)
client.listen()

**Reply by eigenein** [reply]
That one is a known issue: https://github.com/python-smpplib/python-smpplib/issues/105
